core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -267,25 +249,30 @@ public GenericContainer() {
     }
 
     public GenericContainer(@NonNull final String dockerImageName) {
-        this.setDockerImageName(dockerImageName);
+        this(new RemoteDockerImage(DockerImageName.parse(dockerImageName)));
     }
 
     public GenericContainer(@NonNull final Future<String> image) {
-        setImage(image);
+        this(new RemoteDockerImage(image));
+    }
+
+    public GenericContainer(@NonNull final ContainerDef containerDef) {
+        this(containerDef.getImage());
```

Comment:
this will trigger `createContainerDef`. Perhaps we should just do `this.image = containerDef.getImage()`?